components/infra-proxy-service/server/environments.go

```diff
@@ -14,23 +14,104 @@ import (
 	"github.com/chef/automate/components/infra-proxy-service/validation"
 )
 
+// Environment represents the search based deserialized Environment type
+type Environment struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+// EnvironmentResult list result from Search API
+type EnvironmentResult struct {
+	Total int            `json:"total"`
+	Start int            `json:"start"`
+	Rows  []*Environment `json:"rows"`
+}
+
 // GetEnvironments get environments list
 func (s *Server) GetEnvironments(ctx context.Context, req *request.Environments) (*response.Environments, error) {
 	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
 	if err != nil {
 		return nil, err
 	}
 
-	environments, err := c.client.Environments.List()
+	res, err := c.SearchEnvironments(req.SearchQuery)
 	if err != nil {
 		return nil, ParseAPIError(err)
 	}
 
 	return &response.Environments{
-		Environments: fromAPIToListEnvironments(*environments),
+		Environments: fromAPIToListEnvironments(res),
+		Start:        int32(res.Start),
+		Total:        int32(res.Total),
 	}, nil
 }
 
+// SearchEnvironments gets environment list from Chef Infra Server search API.
+func (c *ChefClient) SearchEnvironments(searchQuery *request.SearchQuery) (EnvironmentResult, error) {
+	var result EnvironmentResult
+	var searchAll bool
+	inc := 1000
+	var query chef.SearchQuery
+
+	if searchQuery == nil || searchQuery.Q == "" {
+		searchAll = true
+		query = chef.SearchQuery{
+			Index: "environment",
+			Query: "*:*",
+			Start: 0,
+			Rows:  inc,
+		}
+	} else {
+		perPage := int(searchQuery.GetRows())
+		if perPage == 0 {
+			perPage = 1000
+		}
+
+		query = chef.SearchQuery{
+			Index: "environment",
+			Query: searchQuery.GetQ(),
+			Start: int(searchQuery.GetStart()),
+			Rows:  perPage,
+		}
+	}
+
+	fullURL := fmt.Sprintf("search/%s", query)
+	newReq, err := c.client.NewRequest("GET", fullURL, nil)
+
+	if err != nil {
+		return result, ParseAPIError(err)
+	}
+
+	res, err := c.client.Do(newReq, &result)
+	if err != nil {
+		return result, ParseAPIError(err)
+	}
+
+	defer res.Body.Close() // nolint: errcheck
+
+	if searchAll {
+		var searchResult EnvironmentResult
+		start := result.Start
+		// the total rows available for this query across all pages
+		total := result.Total
+		for start+inc <= total {
+			query.Start = query.Start + inc
+			fullURL = fmt.Sprintf("search/%s", query)
+
+			res1, err := c.client.Do(newReq, &searchResult)
+			if err != nil {
+				return result, ParseAPIError(err)
+			}
+
+			defer res1.Body.Close() // nolint: errcheck
```

Comment:
defer in loop can be problematic. Same comment made on your other pr

Reply:
remove it entirely, not required anymore